apiserver/facades/agent/credentialvalidator/credentialvalidator.go

```diff
@@ -58,35 +59,65 @@ type CloudService interface {
 	Cloud(ctx context.Context, name string) (*cloud.Cloud, error)
 }
 
+// ModelService provides access to the model.
+type ModelService interface {
+	// WatchModelCloudCredential returns a new NotifyWatcher watching for changes that
+	// result in the cloud spec for a model changing. The changes watched for are:
+	// - updates to model cloud.
+	// - updates to model credential.
+	// - changes to the credential set on a model.
+	// The following errors can be expected:
+	// - [modelerrors.NotFound] when the model is not found.
+	WatchModelCloudCredential(ctx context.Context, modelUUID model.UUID) (watcher.NotifyWatcher, error)
+}
+
+// ModelInfoService provides access to the model info.
+type ModelInfoService interface {
+	// GetModelInfo returns the readonly model information for the model in
+	// question.
+	GetModelInfo(ctx context.Context) (model.ModelInfo, error)
+}
+
 type CredentialValidatorAPI struct {
 	*credentialcommon.CredentialManagerAPI
 
-	logger            logger.Logger
-	backend           StateAccessor
-	cloudService      CloudService
-	credentialService CredentialService
-	resources         facade.Resources
+	logger                       logger.Logger
+	cloudService                 CloudService
+	credentialService            CredentialService
+	modelService                 ModelService
+	modelInfoService             ModelInfoService
+	modelCredentialWatcherGetter func(ctx context.Context) (watcher.NotifyWatcher, error)
+	watcherRegistry              facade.WatcherRegistry
+	resources                    facade.Resources
 }
 
 var (
 	_ CredentialValidatorV2 = (*CredentialValidatorAPI)(nil)
 )
 
 func internalNewCredentialValidatorAPI(
-	backend StateAccessor, cloudService CloudService, credentialService CredentialService, resources facade.Resources,
-	authorizer facade.Authorizer, logger logger.Logger,
+	backend credentialcommon.StateBackend, cloudService CloudService, credentialService CredentialService, resources facade.Resources,
```

Comment:
delete `backend credentialcommon.StateBackend`

Reply:
Hi Ian, wanted to clarify on this, dont we need the backend for 
CredentialManagerAPI?

https://github.com/CodingCookieRookie/juju/blob/JUJU-7863-remove-basic-info-agent-credential-validator/apiserver/facades/agent/credentialvalidator/credentialvalidator.go#L112

Reply:
That CredentialManagerAPI plugin just dual writes to state. We can simply take the entire InvalidateModelCredential method and copy it to this facade and drop the state bits and stop using the plugin here. That leaves only one more use of that plugin in the client credential manager facade and so we can fix that in another PR and then delete the plugin entrely

Reply:
Sure thing thanks for the clarifications Ian :D